### PR TITLE
ts/arearange-series

### PR DESCRIPTION
**Pull request by goransle**
Migrated `arearange` series to class pattern.

### CODE REVIEW

**Comment by goransle** [comment]
@bre1470 do you see why the AreaRangeSeries ends up in the `highcharts-more.d.ts` file like this?

```js
/**
 * Adds the module to the imported Highcharts namespace.
 *
 * @param highcharts
 *        The imported Highcharts namespace to extend.
 */
export function factory(highcharts: typeof Highcharts): void;
declare module "./highcharts.src" {

	....

    /**
     * Class
     */
    let AreaRangeSeries: any;
}
....
```

File: ts/Series/AreaRange/AreaRangeDataLabelsOptionsObject.d.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * */
+
+import type DataLabelOptions from '../../Core/Series/DataLabelOptions';
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+export interface AreaRangeDataLabelsOptionsObject extends DataLabelOptions {
```
Comment (bre1470) [inline ts/Series/AreaRange/AreaRangeDataLabelsOptionsObject.d.ts:24]: This all (interface, file name, import, export) should become `AreaRangeDataLabelOptions`.

File: ts/Series/AreaRange/AreaRangePoint.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    public isValid(): boolean {
+        return isNumber(this.low) && isNumber(this.high);
+    }
+}
+
+extend(AreaRangePoint.prototype, {
+
+});
```
Comment (bre1470) [inline ts/Series/AreaRange/AreaRangePoint.ts:177]: You can delete that one. :)

File: ts/Series/AreaSplineRangeSeries.ts [hunk 3]
```diff
@@ -10,8 +10,12 @@
 
 'use strict';
 
+import type AreaRangePoint from './AreaRange/AreaRangePoint';
+import type AreaRangePointOptions from './AreaRange/AreaRangePointOptions';
+import type AreaRangeSeriesOptions from './AreaRange/AreaRangeSeriesOptions';
 import type { SeriesStatesOptions } from '../Core/Series/SeriesOptions';
 import type SplineSeries from './Spline/SplineSeries';
+import AreaRangeSeries from './AreaRange/AreaRangeSeries.js';
```
Comment (bre1470) [inline ts/Series/AreaSplineRangeSeries.ts:18]: `import type` should be enough.

File: ts/Series/DumbbellSeries.ts [hunk 4]
```diff
@@ -38,11 +42,10 @@ const {
...
 } = U;
 
-import './AreaRangeSeries.js';
 import './ColumnRangeSeries.js';
 import '../Core/Interaction.js';
 
-var areaRangeProto = seriesTypes.arearange.prototype,
+var areaRangeProto = AreaRangeSeries.prototype,
     columnRangeProto = seriesTypes.columnrange.prototype,
     areaRangePointProto = areaRangeProto.pointClass.prototype,
```
Comment (bre1470) [inline ts/Series/DumbbellSeries.ts:50]: Maybe it is better to do a deconstruct on BaseSeries here:
```
import BaseSeries from '../Core/Series/Series.js';
const {
    seriesTypes: {
        arearange: {
            prototype: areaRangeProto
        },
        columnrange: {
             prototype: columnRangeProto
        }
    }
} = BaseSeries;
```

**Review by bre1470 — CHANGES_REQUESTED** [review]
Thank you! See comments above and below.

**Review by bre1470 — APPROVED** [review]
Thank you! Looks good. 👍

**Pull request merged**